### PR TITLE
kernel/signal : Change sig_dispatch about delivering the signal to th…

### PR DESCRIPTION
**Pull request by jeongchanKim**
…e specified pid

if pid task/pthread is alive and unblock the signo, it can get the signal first.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [5d26f4fc068925b041a673169d65165461af4a9c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5d26f4fc068925b041a673169d65165461af4a9c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b871f112a50b1d363a60cda55b3b235c36922409] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b871f112a50b1d363a60cda55b3b235c36922409] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [959def4fdf29d55bb772912407a7151c4f7e2bf6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [959def4fdf29d55bb772912407a7151c4f7e2bf6] - Code Rule Check (C++) OK.

File: os/kernel/signal/sig_dispatch.c [hunk 1]
```diff
@@ -440,10 +440,15 @@ int sig_dispatch(pid_t pid, FAR siginfo_t *info)
 
 	stcb = sched_gettcb(pid);
 	if (stcb) {
+
+		if (!sigismember(&stcb->sigprocmask, info->si_signo)) {
```
Comment (sunghan-chang) [inline os/kernel/signal/sig_dispatch.c]: As we discussed on offline, if we should block a signal when we receive it to handle, second one should be pending, should not be passed to another thread.

File: os/kernel/signal/sig_dispatch.c [hunk 2]
```diff
@@ -440,10 +440,15 @@ int sig_dispatch(pid_t pid, FAR siginfo_t *info)
 
 	stcb = sched_gettcb(pid);
 	if (stcb) {
```
Comment (jeongarmy) [inline os/kernel/signal/sig_dispatch.c]: please update description on top of this function about changes.

Comment (jeongchanKim) [reply os/kernel/signal/sig_dispatch.c]: Done.

**Comment by seinfra** [comment]
Target : [2b5a660e74636d97713c2e511ed290517d8f9eea] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2b5a660e74636d97713c2e511ed290517d8f9eea] - Code Rule Check OK.

File: os/kernel/signal/sig_dispatch.c (hunk 1 above)
Comment (davidfather) [reply os/kernel/signal/sig_dispatch.c]: Why don't you commit this change to a new branch for binary separation and IPC?
Signal mechanism modification should be carefully done.

**Comment by seinfra** [comment]
Target : [28687fbc48ff0364f7155616f22eed6ab8f250ca] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [28687fbc48ff0364f7155616f22eed6ab8f250ca] - Code Rule Check (C++) OK.

**Pull request closed without merging**